apps/api/src/queries/photograph/photograph-queries-fetch-many.e2e.spec.ts

```diff
@@ -0,0 +1,403 @@
+                    ]);
+
+                    await testRepositoryProvider
+                        .getContributorRepository()
+                        .create(dummyContributor);
+                });
+
+                it(`should return the unpublished photograph`, async () => {
+                    const res = await request(app.getHttpServer()).get(indexEndpoint);
+
+                    expect(res.status).toBe(httpStatusCodes.ok);
+
+                    const {
+                        body: { entities },
+                    } = res;
+
+                    // this is the first and only photograph we seeded
+                    const result = entities[0] as PhotographViewModel;
+
+                    expect(result.id).toBe(photographIdUnpublishedWithUserAccessId);
+                });
+            });
+        });
+    });
+
+    (
+        [
+            [CoscradUserRole.superAdmin, 'when the user is a COSCRAD admin'],
+            [CoscradUserRole.projectAdmin, 'when the user is a project admin'],
+        ] as const
+    ).forEach(([userRole, description]) => {
+        describe(description, () => {
+            beforeAll(async () => {
+                ({ app, testRepositoryProvider, databaseProvider } = await setUpIntegrationTest(
+                    {
+                        ARANGO_DB_NAME: testDatabaseName,
+                    },
+                    {
+                        testUserWithGroups: new CoscradUserWithGroups(
+                            dummyUser.clone({
+                                roles: [userRole],
+                            }),
+                            []
+                        ),
+                    }
+                ));
+
+                photographQueryRepository = app.get(PHOTOGRAPH_QUERY_REPOSITORY_TOKEN);
+            });
+
+            beforeEach(async () => {
+                await databaseProvider.getDatabaseForCollection('photograph__VIEWS').clear();
+
+                await databaseProvider
+                    .getDatabaseForCollection(ArangoCollectionId.contributors)
+                    .clear();
+
+                await photographQueryRepository.createMany([
+                    publicPhotographView,
+                    privatePhotographThatUserCanAccess,
+                    privatePhotographUserCannotAccess,
+                ]);
+
+                await testRepositoryProvider.getContributorRepository().create(dummyContributor);
+            });
+
+            it(`should allow the user to access private resources`, async () => {
+                const res = await request(app.getHttpServer()).get(indexEndpoint);
+
+                const numberOfPrivatePhotographs = 2;
+
+                const numberOfPublicPhotographs = 1;
+
+                expect(res.status).toBe(HttpStatusCode.ok);
+
+                expect(res.body.entities).toHaveLength(
+                    numberOfPrivatePhotographs + numberOfPublicPhotographs
+                );
+
+                /**
```

Comment:
We should probably find a better test helper pattern. See the comment from the `fetchById` test.